internal/utils.go

```diff
@@ -102,13 +105,51 @@ func getPolicyTemplates(policyConf *types.PolicyConfig) ([]map[string]map[string
 	}
 
 	objectTemplates := make([]map[string]interface{}, 0, len(manifests))
+	policyTemplatesLength := 1
+	if !policyConf.ConsolidateManifests {
+		policyTemplatesLength = len(manifests)
+	}
+	policyTemplates := make([]map[string]map[string]interface{}, 0, policyTemplatesLength)
 	for _, manifest := range manifests {
 		objTemplate := map[string]interface{}{
 			"complianceType":   policyConf.ComplianceType,
 			"objectDefinition": manifest,
 		}
-		objectTemplates = append(objectTemplates, objTemplate)
+		if policyConf.ConsolidateManifests {
+			// put all objTemplate with manifest into single consolidated objectTemplates object
+			objectTemplates = append(objectTemplates, objTemplate)
+		} else {
+			// casting each objTemplate with manifest to objectTemplates type
+			// build policyTemplate for each objectTemplates
+			policyTemplate := buildPolicyTemplate(policyConf, &[]map[string]interface{}{objTemplate})
+			setNamespaceSelector(policyConf, policyTemplate)
+			policyTemplates = append(policyTemplates, *policyTemplate)
+		}
 	}
+
+	//  just build one policyTemplate by using the above consolidated objectTemplates
+	if policyConf.ConsolidateManifests {
+		policyTemplate := buildPolicyTemplate(policyConf, &objectTemplates)
+		setNamespaceSelector(policyConf, policyTemplate)
+		policyTemplates = append(policyTemplates, *policyTemplate)
+	}
+
+	// check the enabled expanders and add additional policy templates
+	expandedPolicyTemplates := handleExpanders(manifests, policyConf)
+	policyTemplates = append(policyTemplates, expandedPolicyTemplates...)
```

Comment:
https://go.dev/blog/slices-intro.  // allocate double what's needed, for future growth.

Just checked that it will double the allocation like (len(policyTemplates) + len(expandedPolicyTemplates)) * 2 here